src/simplify_algebra.cpp

```diff
@@ -78,6 +77,162 @@ struct find_mul_conv
     }
 };
 
+struct find_resize
+{
+    auto matcher() const
+    {
+        return match::name("gather")(
+            match::args(match::name("reshape")(match::arg(0).bind("data")),
+                        match::is_constant().bind("ind")),
+            match::all_of[match::outputs()](match::used_once(), pointwise()));
```

Comment:
I dont think its necessary to check if the outputs are pointwise or that they are used once. This reshape+broadcast transformation should work even if the outputs are convolution.

Reply:
fixed